harvard-lts/mirador-pdiiif-plugin
# Filename updates

**JIRA Ticket**: [LTSVIEWER-197](https://jira.huit.harvard.edu/browse/LTSVIEWER-197)

- [Companion PR](https://github.com/harvard-lts/mps-viewer/pull/53)

# What does this Pull Request do?
1. Fixes an issue where v3 manifests would use the filename `[Object object]` instead of the label from the manifest. This now uses a Mirador selector, which will invoke Manifesto.js to query the manifest label. This could in theory return something other than a string, so has some precautionary handling for that.
2. Limits the filename to 256 characters

# How should this be tested?

1. Checkout the [companion PR](https://github.com/harvard-lts/mps-viewer/pull/53) for mps-viewer
2. Restart your local docker container for the viewer. This should install the newest dependencies and run webpack
3. Navigate to a V2 PTO manifest (e.g. one of the legacy PTO's)
4. Saving the file should produce a file using the label field from the manifest (try both Chrome and Firefox)
5. The filename (excluding ".pdf") should not exceed 256 characters (in practice this doesn't apply to any Harvard manifest I've tested)
6. Try with one of the V3 manifests from the QA matrix (One which doesn't require the VPN). The filename should use the label field as with the legacy manifests

# Test coverage

Yes/No: Are changes in this pull-request covered by:

- unit tests? No
- integration tests? No

# Interested parties
@enriquediaz
